actors/verifreg/src/lib.rs

```diff
@@ -366,108 +366,104 @@ impl Actor {
-                };
-
-                let inserted = claims
-                    .put_if_absent(provider, claim_alloc.allocation_id, new_claim)
-                    .context_code(
+                // Update state.
+                // Errors from here on are unexpected, so abort.
+                let mut sector_claimed_space = DataCap::zero();
+                for (id, new_claim) in sector_new_claims {
+                    let inserted =
+                        claims.put_if_absent(provider, id, new_claim.clone()).context_code(
+                            ExitCode::USR_ILLEGAL_STATE,
+                            format!("failed to write claim {}", id),
+                        )?;
+                    if !inserted {
+                        return Err(actor_error!(illegal_argument, "claim {} already exists", id));
+                    }
+                    allocs.remove(new_claim.client, id).context_code(
                         ExitCode::USR_ILLEGAL_STATE,
-                        format!("failed to write claim {}", claim_alloc.allocation_id),
+                        format!("failed to remove allocation {}", id),
                     )?;
-                if !inserted {
-                    ret_gen.add_fail(ExitCode::USR_ILLEGAL_STATE);
-                    // should be unreachable since claim and alloc can't exist at once
-                    info!(
-                        "claim for allocation {} could not be inserted as it already exists",
-                        claim_alloc.allocation_id,
-                    );
-                    continue;
+                    sector_claimed_space += DataCap::from(new_claim.size.0);
                 }
-
-                allocs.remove(claim_alloc.client, claim_alloc.allocation_id).context_code(
-                    ExitCode::USR_ILLEGAL_STATE,
-                    format!("failed to remove allocation {}", claim_alloc.allocation_id),
-                )?;
-
-                total_datacap_claimed += DataCap::from(claim_alloc.size.0);
-                sector_claims
-                    .push(SectorAllocationClaimResult { claimed_space: claim_alloc.size.0.into() });
-                ret_gen.add_success();
+                total_claimed_space += &sector_claimed_space;
+                sector_results.push(SectorAllocationClaim { claimed_space: sector_claimed_space });
+                batch_gen.add_success();
             }
             st.save_allocs(&mut allocs)?;
             st.save_claims(&mut claims)?;
             Ok(())
         })
         .context("state transaction failed")?;
-        let batch_info = ret_gen.gen();
+
+        let batch_info = batch_gen.gen();
         if params.all_or_nothing && !batch_info.all_ok() {
-            return Err(actor_error!(
-                illegal_argument,
-                "all or nothing call contained failures: {}",
-                batch_info.to_string()
+            return Err(ActorError::checked(
+                batch_info.fail_codes[0].code,
```

Comment:
This change to propagate the exit code from the firs failing item when all_or_nothing is true, rather than illegal argument, caused the integration test failure. I will revert it, though think perhaps propagating the code is better.